### PR TITLE
[BUILD] Execute stf tests on pr/stf/* branches

### PR DESCRIPTION
**Pull request by m273d15**
* Introduce GitHub action in order to execute
  the stf tests on branches that improve the
  stf tests.
* The travis stf runs terminate after ~45min
  and don't allow to run the tests completely

### CODE REVIEW

**Comment by srossbach** [comment]
Any changes to only execute modified or new tests instead of all tests ?

**Comment by m273d15** [comment]
No I want to create a setup that allows to modify the tests in pr/stf branches.
The Travis setup terminates the test and only runs nightly. Follow-up PRs will change the set of stf test to be executed

File: .github/workflows/stf.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+name: STF Tests
+
+on:
+  push:
+    # Temporary restriction: Run only stf tests on branches that work on stf
+    branches:    
+      - 'pr/stf/*'
+
+# Restriction when STFs ha stable results
```
Comment (tobous) [inline .github/workflows/stf.yml]:
```suggestion
# Restriction when STF has stable results
```

**Comment by m273d15** [comment]
> The action fails with an exception regarding the yaml file:

I fixed this with a direct rebase push. The indention of the file was broken. It is the parse that failed not the action. The [GitHub documentation](https://help.github.com/en/github/setting-up-and-managing-organizations-and-teams/repository-permission-levels-for-an-organization#repository-access-for-each-permission-level) says that you need at least "write" access in order to create/edit an action.

**Comment by tobous** [comment]
> The [GitHub documentation](https://help.github.com/en/github/setting-up-and-managing-organizations-and-teams/repository-permission-levels-for-an-organization#repository-access-for-each-permission-level) says that you need at least "write" access in order to create/edit an action.

Ok, good.

> I fixed this with a direct rebase push. The indentation of the file was broken. It is the parse that failed not the action.

Is there a way to re-trigger the action so that we can make sure that it runs correctly before merging this?

**Comment by m273d15** [comment]
I created the branch `pr/stf/test_gha` with the same commits. Therefore, the result is also published on this PR, because the same commit is build. The result: STF GitHub Action badge is as expected, because of #744

File: .github/workflows/stf.yml (hunk 1 above)
Comment (tobous) [reply .github/workflows/stf.yml]: Still unresolved.

**Comment by tobous** [comment]
It still feels kind of weird adding a check that we know will always fail in the current setup, meaning the builds that trigger them will always fail (until all issues with the current STF tests are resolved). But I guess that's how it'll be for now. At least having any job that runs the STF completely is an improvement.

**Pull request merged**